### PR TITLE
Return the group name when listing a user's groups.

### LINKED ISSUE #1418: ReBAC: User's group names are ids
In the response for a user's groups: `/identities/{id}/groups` the name value contains the id:

```json
{
    "_links": {
        "next": {
            "href": ""
        }
    },
    "_meta": {
        "pageToken": "",
        "size": 1
    },
    "data": [
        {
            "id": "a084509c-33c8-4ec7-b46c-d0f74958f892",
            "name": "a084509c-33c8-4ec7-b46c-d0f74958f892"
        }
    ],
    "message": "",
    "status": 200
}
```

The same group from the `/groups` response is:

```
{
    "_links": {
        "next": {
            "href": "/rebac/v1/groups?page=1"
        }
    },
    "_meta": {
        "page": 0,
        "size": 1,
        "total": 1
    },
    "data": [
        {
            "id": "a084509c-33c8-4ec7-b46c-d0f74958f892",
            "name": "group1"
        }
    ],
    "message": "",
    "status": 200
}
```

### PR DESCRIPTION
**Pull request by kian99**
## Description

This PR fixes the response from the rebac admin's "GetIdentityGroups" endpoint. Currently we return a group name in place of the group ID. One issue we face is that the group name may no longer exist in the DB but an OpenFGA may still exist. To handle this case I considered:
- Returning an error
- Dropping the offending result
- Returning the group with a name of "removed"

~~I've opted for option 3. The first would prevent the user from viewing their groups and the second would hide the issue.~~
Changed to option 2 to drop the result since it is not useful to the user.

Fixes #1418.

## Engineering checklist
*Check only items that apply*

- [ ] Documentation updated
- [x] Covered by unit tests
- [x] Covered by integration tests

### CODE REVIEW

File: internal/jimmhttp/rebac_admin/identities.go
Comment (SimoneDutto) [inline internal/jimmhttp/rebac_admin/identities.go:129]: i think since we already have all the ids we could create a `GetGroupsByUUIDs`.
Otherwise this name resolution could potentially be a performance hit.
What do you think?

File: internal/jimmhttp/rebac_admin/identities.go
Comment (SimoneDutto) [inline internal/jimmhttp/rebac_admin/identities.go]: thinking out loud...
if the group is missing from the db it won't be listed in the `/groups | group/{id}` endpoint, so showing the id in this endpoint doesn't make sense, because it can't be resolved if the user clicks on it.

So I would remove these entries missing from postgres, so the user doesn't see "dangling" uuid.
Just a `continue` without returning any error is fine.

Comment (kian99) [reply internal/jimmhttp/rebac_admin/identities.go]: I agree, dropping the result does seem like the better thing to do. I'll wait for another review to chime in and if we agree, I'll change it.

File: internal/jimmhttp/rebac_admin/identities.go
Comment (kian99) [reply internal/jimmhttp/rebac_admin/identities.go:129]: It could but I didn't want to optimise prematurely. Curious to hear other opinions. If it becomes an issue we can fix it later? Makes me wonder if we have monitoring on our Rebac-admin endpoints.

Comment (SimoneDutto) [reply internal/jimmhttp/rebac_admin/identities.go:129]: We have, i've added the middleware monitoring in this pr: https://github.com/canonical/jimm/pull/1412

File: internal/jimmhttp/rebac_admin/identities.go
Comment (kian99) [reply internal/jimmhttp/rebac_admin/identities.go]: Changed it.

File: internal/jimmhttp/rebac_admin/identities.go
Comment (SimoneDutto) [inline internal/jimmhttp/rebac_admin/identities.go]: you've missed to update the comment

File: internal/jimmhttp/rebac_admin/identities_integration_test.go
Comment (ale8k) [inline internal/jimmhttp/rebac_admin/identities_integration_test.go]: Don't need this

File: internal/jimmhttp/rebac_admin/identities_integration_test.go
Comment (ale8k) [inline internal/jimmhttp/rebac_admin/identities_integration_test.go]: TestGetIdentityGroupsWithDeletedDbGroup would be better (just because the method is named wrong)

File: internal/jimmhttp/rebac_admin/identities.go
Comment (ale8k) [inline internal/jimmhttp/rebac_admin/identities.go:129]: Would be nicer GetGroupsByUUIDs but it doesn't matter much

File: internal/jimmhttp/rebac_admin/identities.go
Comment (ale8k) [inline internal/jimmhttp/rebac_admin/identities.go]: How is the group shown as "removed"? We just call .First

File: internal/jimmhttp/rebac_admin/identities.go
Comment (ale8k) [inline internal/jimmhttp/rebac_admin/identities.go:133]: Should we be leaving lingering tuples? Or would .GetGroupsByUUIDs be better in the JIMM layer with tuple cleanup? (Really not sure here just opening up for discussion)

**Review by ale8k — CHANGES_REQUESTED** [review]

**Review by ale8k — CHANGES_REQUESTED** [review]
Requesting changes on test name and some comments need answering before I'm comfortable to approve.

File: internal/jimmhttp/rebac_admin/identities.go
Comment (alesstimec) [reply internal/jimmhttp/rebac_admin/identities.go:129]: since we already have the GetGroupByUUID, it's ok. agree with Kian here.

File: internal/jimmhttp/rebac_admin/identities.go
Comment (alesstimec) [reply internal/jimmhttp/rebac_admin/identities.go]: i think cleanup could possibly be a topic in its own right. we leave a lot of tuples in openfga that refer to a deleted entity in jimm. i think a spec on this would be very welcome, if anybody has time.

**Review by alesstimec — DISMISSED** [review]
LGTM with comments addressed

File: internal/jimmhttp/rebac_admin/identities.go
Comment (kian99) [reply internal/jimmhttp/rebac_admin/identities.go]: Comment was outdated, we just drop the group from the result.

File: internal/jimmhttp/rebac_admin/identities.go
Comment (kian99) [reply internal/jimmhttp/rebac_admin/identities.go:133]: We shouldn't be leaving lingering tuples, but that's a separate issue. This is just handling the case that, for some reason, there is a leftover tuple but no group in the database.

File: internal/jimmhttp/rebac_admin/identities_integration_test.go
Comment (alesstimec) [inline internal/jimmhttp/rebac_admin/identities_integration_test.go:144]: i don't think you need to get group2 again.. you have its tag on L126 (which contains the id).

**Review by alesstimec — APPROVED** [review]

File: internal/jimmhttp/rebac_admin/identities_integration_test.go
Comment (kian99) [reply internal/jimmhttp/rebac_admin/identities_integration_test.go:144]: I thought so too, but the delete method requires the ID or UUID.

File: internal/jimmhttp/rebac_admin/identities.go
Comment (ale8k) [reply internal/jimmhttp/rebac_admin/identities.go:133]: Not sure what you mean here but perhaps explain in tomorrows standup or offline in element?

File: internal/jimmhttp/rebac_admin/identities.go
Comment (ale8k) [inline internal/jimmhttp/rebac_admin/identities.go:132]: We need to handle the lingering tuples... Should have a discussion on this

**Review by ale8k — APPROVED** [review]

**Review by ale8k — APPROVED** [review]
lgtm bar our lingering naughty tuple issues

**Pull request merged**
